pkg/core/klinedriver/bucket.go

```diff
@@ -0,0 +1,112 @@
+package klinedriver
+
+import (
+	"time"
+
+	"github.com/c9s/bbgo/pkg/fixedpoint"
+	"github.com/c9s/bbgo/pkg/types"
+)
+
+type Bucket struct {
+	// StartTime is inclusive, EndTime is exclusive
+	// e.g. [StartTime, EndTime)
+	StartTime, EndTime types.Time
+	Interval           types.Interval
+	KLine              *types.KLine
+	Exposed            bool
+}
+
+// check if the currentTime is in [startTime, startTime + interval)
+func (bucket *Bucket) Contains(currentTime types.Time) bool {
+	if currentTime.Equal(bucket.StartTime.Time()) {
```

Comment:
(optional)
Let's add a function EqualAfter in types.Time to make the code more readable

Reply:
It's kinda minor to me.
I'd rather to keep this PR more focused.